src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java

```diff
@@ -401,7 +399,71 @@ void consumeRecords(final List<O> recOps) {
     }
   }
 
-  void createTopics(final List<O> topicOps) {}
+  void createTopics(final List<O> topicOps) {
+    val nodeAddr = topicOps.get(0).nodeAddr();
+    try {
+      val config = configCache.computeIfAbsent(nodeAddr, this::createAdminClientConfig);
+      val adminClientCreateFunc =
+          adminClientCreateFuncCache.computeIfAbsent(config, AdminClientCreateFunctionImpl::new);
+      val adminClient = adminClientCache.computeIfAbsent(nodeAddr, adminClientCreateFunc);
+      val topicCollection = new ArrayList<NewTopic>();
+      for (var i = 0; i < topicOps.size(); i++) {
+        val topicOp = topicOps.get(i);
+        val topicName = topicOp.item().name();
+        val newTopic = new NewTopic(topicName, 1, (short) 1);
+        topicCollection.add(newTopic);
+      }
+      concurrencyThrottle.acquire();
+      val createTopicsResult = adminClient.createTopics(topicCollection);
+      for (var i = 0; i < topicOps.size(); i++) {
+        val topicOp = topicOps.get(i);
+        val topicName = topicOp.item().name();
+        try {
+          topicOp.startRequest();
+          val oneTopicResult = createTopicsResult.values().get(topicName);
+          oneTopicResult.whenComplete(
```

Comment:
Consider the refactoring: move the callback code into the separate method call, something like `completeTopicCreate`